dask/array/core.py

```diff
@@ -2982,7 +2984,11 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
 
     n = len(seq)
     if n == 0:
-        return from_array(np.empty(shape=shape, dtype=meta.dtype))
+        try:
+            return wrap.empty_like(meta, shape=shape, chunks=shape,
+                                   dtype=meta.dtype)
+        except TypeError:
+            return wrap.empty(shape, chunks=shape, dtype=meta.dtype)
```

Comment:
You might want `dask.array.utils.empty_like_safe`.  I think that that is what @pentschev has been using in cases like these.

Reply:
That would work here if we can skip `chunks=` for now (as in the behavior prior to this PR). Otherwise we need to add `shape=` to Dask's `*_like` functions, which isn't done yet. I can work on that PR if it's of immediate help.

Reply:
> You might want `dask.array.utils.empty_like_safe`.

It looked like that was [just operating on NumPy arrays]( https://github.com/dask/dask/blob/f7d73f8f448c761b5e7e818835f8aa00057de812/dask/array/utils.py#L221-L230 ). Am I missing something?

Reply:
Correct, this is why I said that can be used if there's no need to pass `chunks`. I guess you need a Dask array here?

Reply:
That's right. We need a Dask Array here.

Reply:
Then I'd say we go ahead with this, and use `da.empty_like` once we have `shape` there. I'll try to add that soon.